spinalcordtoolbox/straightening.py

```diff
@@ -485,13 +484,34 @@ def straighten(self):
 
                 data_warp_straight2curved[indexes[:, 0], indexes[:, 1], indexes[:, 2], 0, :] = -displacements_curved
 
+        # If we're generating a curved2straight warping field, we can update the boundaries of the straight-space safe
+        # zone by finding out which straight-space indices map to the "safe zone" curved-space indices.
+        # FIXME: This should be possible using `lookup_curved2straight` instead, but for some reason, half of that
+        #        lookup table is erroneously set to 0 instead
+        if self.curved2straight:
+            coord_bound_straight_inferior = list(lookup_straight2curved).index(inferior_bound)
+            coord_bound_straight_superior = list(lookup_straight2curved).index(superior_bound)
+            z_centerline_straight = centerline_straight.points[:, 2]
+            bound_straight = [z_centerline_straight[coord_bound_straight_inferior],
+                              z_centerline_straight[coord_bound_straight_superior]]
+            logger.info('Safe zone boundaries (straight space): {}'.format(bound_straight))
+        # If we're not generating a curved2straight warping field, the straight-space safe zone will have no purpose
+        else:
+            bound_straight = [0, 0]
+
         # Creation of the safe zone based on pre-calculated safe boundaries
         coord_bound_curved_inf, coord_bound_curved_sup = image_centerline_pad.transfo_phys2pix(
             [[0, 0, bound_curved[0]]]), image_centerline_pad.transfo_phys2pix([[0, 0, bound_curved[1]]])
         coord_bound_straight_inf, coord_bound_straight_sup = image_centerline_straight.transfo_phys2pix(
             [[0, 0, bound_straight[0]]]), image_centerline_straight.transfo_phys2pix([[0, 0, bound_straight[1]]])
 
-        if radius_safe > 0:
+        # note: we need to ensure that straight bounds don't go outside the warping field's z-dimension, since
+        #       `transfo_phys2pix` has no safeguards, and could return negative or too large indices.
+        coord_bound_straight_inf[0][2] = max(0, coord_bound_straight_inf[0][2])
+        coord_bound_straight_sup[0][2] = min(nz_s, coord_bound_straight_sup[0][2])
+
+        if self.safe_zone:
+            logger.info('Applying safe zone to warping fields')
```

Comment:
Due to the new `batch_processing.sh` "detailed check", we can see that this change did in fact affect the warping field produced during the T2 registration to template: https://github.com/spinalcordtoolbox/spinalcordtoolbox/actions/runs/15640363898/job/44068537856

This in turn had a small effect on all of the files warped from the template to the T2 space. However, usefully, this was not enough to change the results of `batch_processing.sh`, nor was it enough to change the effect on the subsequent registration steps (where the warping fields are reused for `-initwarp`, etc.).

While not an entirely thorough test, it is at least comforting, and is enough for me to mark the code changes as ready for review.

Reply:
On closer look, it looks like my changes may have inadvertently applied a `100000` value instead of the `-100000` value that the [`threshold_distance`](https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/4884#issuecomment-2968356463) code applies: https://github.com/spinalcordtoolbox/spinalcordtoolbox/actions/runs/15640363898/job/44068537856#step:5:2170

This likely isn't a huge difference, but I'm curious if keeping the value consistent will have a smaller effect on the output warping field.

Reply:
It looks like there may be an off-by-one error? When I check the `batch_processing.sh` results, the top slice of the warping field gets overwritten on my branch but is preserved on `master`, despite the spinal cord covering that slice.

That would explain the slight differences. I'll try to dig in deeper. :)

Reply:
Hrm. Fixing this off-by-one error reduced the number of changed properties to ~70. Now the `warp_curve2straight.nii.gz` warping field is identical, but the `warp_straight2curve.nii.gz` warping field differs.

I'll try to dig into the changes caused to that warping field, too. :)